mbway/src/main/java/com/adyen/checkout/mbway/MBWayComponent.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 11/8/2020.
+ */
+package com.adyen.checkout.mbway
+
+import com.adyen.checkout.components.PaymentComponentProvider
+import com.adyen.checkout.components.PaymentComponentState
+import com.adyen.checkout.components.base.BasePaymentComponent
+import com.adyen.checkout.components.base.GenericPaymentComponentProvider
+import com.adyen.checkout.components.base.GenericPaymentMethodDelegate
+import com.adyen.checkout.components.model.payments.request.MBWayPaymentMethod
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+
+/**
+ * Component should not be instantiated directly. Instead use the PROVIDER object.
+ *
+ * @param paymentMethodDelegate [GenericPaymentMethodDelegate]
+ * @param configuration [MBWayConfiguration]
+ */
+class MBWayComponent(paymentMethodDelegate: GenericPaymentMethodDelegate, configuration: MBWayConfiguration) :
+    BasePaymentComponent<MBWayConfiguration, MBWayInputData, MBWayOutputData,
+        PaymentComponentState<MBWayPaymentMethod>>(paymentMethodDelegate, configuration) {
+
+    companion object {
+        private val TAG = LogUtil.getTag()
+        val PROVIDER: PaymentComponentProvider<MBWayComponent, MBWayConfiguration> = GenericPaymentComponentProvider(MBWayComponent::class.java)
```

Comment:
Make sure `PROVIDER` is `@JvmStatic`, we still have a lot of merchants that use Java.

Also, for variables that are private, do you think it's better to have them outside the class as top level private instead of the companion object? I don't have a strong opinion about this, just curious.